doc: Add README for basic, basic_lorawan, freefall_lorawan apps (#195)
TheThingsIndustries/generic-node-se: Software/app/freefall_lorawan/README.md

# freefall\_lorawan

## Description

This application is used to detect free fall events and also serves as a template for acceleration detection handling. Each time a free fall event is detected by the accelerometer, a message containing a counter in the payload is sent. The transmitted value is incremented each time the event occurs. Additionally, if a downlink is received over the correct port, which can be configured by the user, the buzzer will beep periodically to notify locally that the GNSE should be picked up. The buzzer can be turned off by pressing the user button.

## Configuration

### Default
1. Follow [this](https://www.thethingsindustries.com/docs/devices/adding-devices/) guide to add your device to TTN.
2. Change `APPEUI`, `DEVEUI` and `APPKEY` in [conf/app_conf.h](https://github.com/TheThingsIndustries/generic-node-se/blob/develop/Software/app/freefall_lorawan/conf/app_conf.h) to the values for your specific device on TTN.

> 3. Uncomment the REGION parameters in [conf/lorawan_conf.h](https://github.com/TheThingsIndustries/generic-node-se/blob/develop/Software/app/freefall_lorawan/conf/lorawan_conf.h) and set `ACTIVE_REGION` in [lora_app.h](https://github.com/TheThingsIndustries/generic-node-se/blob/develop/Software/app/freefall_lorawan/lora_app.h). These should correspond to the location that you use these devices for.

Reviewer: Same as in the previous file

### Free fall detection

> 1. Change `ACC_FF_THRESHOLD` in [app/conf.h](https://github.com/TheThingsIndustries/generic-node-se/blob/develop/Software/app/freefall_lorawan/conf/app_conf.h) to your liking. This will determine the threshold where all axis can be from 0g (which would indicate that it is free falling) to be considered a free fall event.

Reviewer: ```suggestion
1. Change `ACC_FF_THRESHOLD` in [app/conf.h](https://github.com/TheThingsIndustries/generic-node-se/blob/develop/Software/app/freefall_lorawan/conf/app_conf.h) to a value you prefer. This will determine the threshold where all axis can be from 0g (which would indicate that it is free falling) to be considered a free fall event.
```

> 2. Change `ACC_FF_DURATION` in [app/conf.h](https://github.com/TheThingsIndustries/generic-node-se/blob/develop/Software/app/freefall_lorawan/conf/app_conf.h) to your liking. This will determine how many times the axis have to meet the threshold values. Note that `ACC_FF_ODR` in the same file also has influence on this.

Reviewer: ```suggestion
2. Change `ACC_FF_DURATION` in [app/conf.h](https://github.com/TheThingsIndustries/generic-node-se/blob/develop/Software/app/freefall_lorawan/conf/app_conf.h) to a value you prefer. This will determine how many times the axis have to meet the threshold values. Note that `ACC_FF_ODR` in the same file also affects this.
```

3. `ACC_FF_LORA_PORT` can be changed in [conf/app_conf.h](https://github.com/TheThingsIndustries/generic-node-se/blob/develop/Software/app/freefall_lorawan/conf/app_conf.h), which is used to configure to transmission port. The LoRaWAN keys mentioned in the default section can be altered here as well.
4. Also in [conf/app_conf.h](https://github.com/TheThingsIndustries/generic-node-se/blob/develop/Software/app/freefall_lorawan/conf/app_conf.h), the downlink port can be set by altering `ACC_FF_DOWNLINK_PORT`.

### Debugger

For debugging, the firmware has to support it first. The debugger is set in the macro `DEBUGGER_ON` in [conf/app_conf.h](https://github.com/TheThingsIndustries/generic-node-se/blob/develop/Software/app/freefall_lorawan/conf/app_conf.h).

> 

Reviewer: This is pretty much the same in both files, so please look at the comments for the first one

## Setup

> 1. See if your device can detect free fall events. The default values should be enough to trigger on minor drops (try to not break the device) or even sudden movements with your arm whilst holding the GNSE. A message should be transmitted over LoRaWAN containing the counter. If none can be seen, try to debug the device as described in the Observation section.

Reviewer: ```suggestion
1. See if your device can detect free fall events. The default values should be enough to trigger on minor drops (try to not break the device) or even sudden movements with your arm whilst holding the GNSE. A message containing the counter should be transmitted. If none can be seen, try to debug the device as described in the Observation section.
```

Reviewer: No need to highlight that it's LoRaWAN, I think that's more than obvious. Also, link the Observation section

> 2. Test you setup for free fall events and change it to your liking (with the steps in the Configuration chapter). Be aware that `ACC_FF_THRESHOLD` would have to be increased to increase the change of a free fall event whilst `ACC_FF_DURATION` would have to be decreased for a similar effect.

Reviewer: Link the Configuration section and remove the word "chapter". What does "to increase the change of a free fall event" mean? Please reword this a bit

## Observation

> On TTN, the payload should be a single byte starting at one. When a downlink is received on the port set in `ACC_FF_DOWNLINK_PORT` in [app/conf.h](https://github.com/TheThingsIndustries/generic-node-se/blob/develop/Software/app/freefall_lorawan/conf/app_conf.h), then the buzzer should make noises.

Reviewer: ```suggestion
On The Things Stack, the payload should be a single byte starting at one. When a downlink is received on the port set in `ACC_FF_DOWNLINK_PORT` in [app/conf.h](https://github.com/TheThingsIndustries/generic-node-se/blob/develop/Software/app/freefall_lorawan/conf/app_conf.h), then the buzzer should start make noises.
```

Reviewer: What does "a single byte starting at one" mean? What is "one" here?

UART on the TX/RX lines can also be used, which are configured to 115200/8-N-1 by default. If you are having troubles with the device, check if the UART output matches these things:

1. The accelerometer is initialized to detect free fall events, which is notified with this message:
```
Accelerometer initialized

> ```

Reviewer: Add blank lines above and below, same for text below

> 2. The correct keys are used, which are printed at the start of the application.

Reviewer: ```suggestion
2. The correct keys are used, i.e. they match the ones printed at the start of the application.
```

3. The device is connecting properly, as it will indicate if connections have failed or not. It should display something like this:
```
###### = JOINED = OTAA =====================
```
4. The device is transmitting anything, since each event is also logged via UART. It should display something similar to this:
```
###### ========== MCPS-Confirm =============
###### U/L FRAME:0005 | PORT:2 | DR:5 | PWR:0 | MSG TYPE:UNCONFIRMED
60s069:TX on freq 868500000 Hz at DR 5
60s080:SEND REQUEST
```

> 5. Downlinks are detected if used. These only occur after being explicitly set in the TTN console or another application, and will only be transmitted after the transmission of an uplink by the device. When any data is set to the port set in `ACC_FF_DOWNLINK_PORT` in [app/conf.h](https://github.com/TheThingsIndustries/generic-node-se/blob/develop/Software/app/freefall_lorawan/conf/app_conf.h) then the buzzer can also be heard. Additionally, the GNSE should, after receiving the downlink, display something similar to this:

Reviewer: ```suggestion
5. Downlinks are detected only if used. These only occur after being explicitly set in the The Things Stack Console or another application, and will only be transmitted after the transmission of an uplink. When any data is sent to the port set in `ACC_FF_DOWNLINK_PORT` in [app/conf.h](https://github.com/TheThingsIndustries/generic-node-se/blob/develop/Software/app/freefall_lorawan/conf/app_conf.h) then the buzzer can also be heard. Additionally, after receiving the downlink, the GNSE should display something similar to:
```
